### PR TITLE
Add Data.Streaming.ByteString.Builder

### PR DESCRIPTION
**Pull request by manny-fp**

### CODE REVIEW

File: streaming-commons.cabal [hunk 1]
```diff
@@ -65,12 +66,16 @@ library
   else
     build-depends:     unix
 
+  if impl(ghc < 7.7.0)
```
Comment (snoyberg) [inline streaming-commons.cabal]: This shouldn't be based on the GHC version, but rather on the bytestring version.

**Comment by snoyberg** [comment]
Note the Travis build failures, looks like the test suite needs a bit of conditional logic added.

**Comment by manny-fp** [comment]
@snoyberg Travis now passes.  I missed that because I was also testing different GHC versions with `--enable-benchmarks`, and one of the benchmarks dependencies pulled in bytestring-0.10.4.x.

**Comment by snoyberg** [comment]
W00t! Looks good.

**Pull request merged**